src/Language/PureScript/Errors.hs

```diff
@@ -500,33 +496,41 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage . positionHintsFir
     renderSimpleErrorMessage (ConflictingCtorDecls nm) =
       line $ "Conflicting data constructor declarations for " ++ runProperName nm
     renderSimpleErrorMessage (TypeConflictsWithClass nm) =
-      line $ "Type " ++ runProperName nm ++ " conflicts with type class declaration of the same name"
+      line $ "The type " ++ runProperName nm ++ " conflicts with a type class declaration of the same name."
     renderSimpleErrorMessage (CtorConflictsWithClass nm) =
-      line $ "Data constructor " ++ runProperName nm ++ " conflicts with type class declaration of the same name"
+      line $ "The data constructor " ++ runProperName nm ++ " conflicts with a type class declaration of the same name."
     renderSimpleErrorMessage (ClassConflictsWithType nm) =
-      line $ "Type class " ++ runProperName nm ++ " conflicts with type declaration of the same name"
+      line $ "The type class " ++ runProperName nm ++ " conflicts with a type declaration of the same name."
     renderSimpleErrorMessage (ClassConflictsWithCtor nm) =
-      line $ "Type class " ++ runProperName nm ++ " conflicts with data constructor declaration of the same name"
+      line $ "The type class " ++ runProperName nm ++ " conflicts with a data constructor declaration of the same name."
     renderSimpleErrorMessage (DuplicateModuleName mn) =
-      line $ "Module " ++ runModuleName mn ++ " has been defined multiple times."
+      line $ "The module " ++ runModuleName mn ++ " has been defined multiple times."
     renderSimpleErrorMessage (DuplicateClassExport nm) =
       line $ "Duplicate export declaration for type class " ++ runProperName nm
     renderSimpleErrorMessage (DuplicateValueExport nm) =
       line $ "Duplicate export declaration for value " ++ showIdent nm
     renderSimpleErrorMessage (CycleInDeclaration nm) =
-      line $ "Cycle in declaration of " ++ showIdent nm
+      paras [ line $ "The declaration of " ++ showIdent nm ++ " contains a cycle."
+            , line $ "The value of " ++ showIdent nm ++ " may not be defined here, so this reference is not allowed."
```

Comment:
I'm not sure this second line helps - it's just a reiteration of the error, I think just:

```
The declaration of x contains a cycle.
You might be able to evaluate this expression lazily, using something like Control.Lazy.fix
```

Would be better. Or if you think further elaboration is needed state something about the declaration being infinitely recursive / non-terminating.